### PR TITLE
[Test] Refactor `inject_webhook_test.go` according to the Developer Guide

### LINKED ISSUE #2388: Refactor `/pkg/webhook/v1beta1/pod/inject_webhook_test.go` according to Developer Guide.
### What you would like to be added?

Refactor `/pkg/webhook/v1beta1/pod/inject_webhook_test.go` to obey rules defined in the [Developer Guide](https://github.com/kubeflow/katib/blob/master/docs/developer-guide.md).

### Why is this needed?

Currently, testcases in `/pkg/webhook/v1beta1/pod/inject_webhook_test.go` are defined in slices and use `reflect.Equal` for comparison. It's not recommended according to the Developer Guide. We should:

1. Use [cmp.Diff](https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff) instead of reflect.Equal, to provide useful comparisons.

2. Define test cases as maps instead of slices to avoid dependencies on the running order. Map key should be equal to the test case name.

### Love this feature?

Give it a 👍 We prioritize the features with most 👍

### PR DESCRIPTION
**Pull request by Electronic-Waste**
**What this PR does / why we need it**:

I refactored `/pkg/webhook/v1beta1/pod/inject_webhook_test.go` with rule described in [Developer Guide](https://github.com/kubeflow/katib/blob/master/docs/developer-guide.md):

1. Use [cmp.Diff](https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff) instead of reflect.Equal, to provide useful comparisons.
2. Define test cases as maps instead of slices to avoid dependencies on the running order. Map key should be equal to the test case name.

**Which issue(s) this PR fixes** _(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)_:
Fixes #2388

**Checklist:**

- [ ] [Docs](https://www.kubeflow.org/docs/components/katib/) included if any changes are user facing

### CODE REVIEW

File: pkg/webhook/v1beta1/pod/inject_webhook_test.go [hunk 1]
```diff
@@ -510,23 +496,6 @@ func TestGetMetricsCollectorArgs(t *testing.T) {
...
-			mCSpec: common.MetricsCollectorSpec{
-				Collector: &common.CollectorSpec{
-					Kind: common.StdOutCollector,
-				},
-			},
-			earlyStoppingRules: earlyStoppingRules,
-			katibConfig:        configv1beta1.MetricsCollectorConfig{},
-			name:               "Trial with invalid Experiment label name. Suggestion is not created",
-			err:                true,
 		},
```
Comment (tenzen-y) [inline pkg/webhook/v1beta1/pod/inject_webhook_test.go:499]: What is the reason that we need to remove this case?

Comment (Electronic-Waste) [reply pkg/webhook/v1beta1/pod/inject_webhook_test.go:499]: Sorry,I might delete this test case by accident.

**Comment by Electronic-Waste** [comment]
I've added the missing UT which was deleted by accident now. PTAL👀 @tenzen-y  if you have time.

**Review by tenzen-y — COMMENTED** [review]
Thanks!
/lgtm
/approve

**Pull request merged**